types/react-transition-group/Transition.d.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, HTMLProps } from "react";
+import { TransitionActions } from "react-transition-group";
+
+export type EndHandler = (node: HTMLElement, done: () => void) => void;
+export type EnterHandler = (node: HTMLElement, isAppearing: boolean) => void;
+export type ExitHandler = (node: HTMLElement) => void;
+
+export interface TransitionProps extends TransitionActions, HTMLProps<any> {
```

Comment:
`TransitionProps` no longer accepts `HTMLProps<any>`. Rather, it passes through all its additional props to the child by doing a `cloneElement` call, or by calling the `child` function.

There is no need to allow passing arbitrary props in the `Transition` instance anymore.